### PR TITLE
Add -c option that use the current session and spawn a new window

### PR DESCRIPTION
**Pull request by jsecchiero**
For me that i already use tmux in my terminal i prefer to spawn a new window instead of a new session

### CODE REVIEW

**Comment by oryband** [comment]
Hi @peikk0 . Can we get this merged? This is incredibly useful!

**Comment by oryband** [comment]
@jsecchiero I think it's missing an option to name the new window

**Comment by jsecchiero** [comment]
Yep i think we can use the same -n switch used for new sessions

**Comment by jsecchiero** [comment]
now with -n (and -c for use the current session) you can choose the window name.
@oryband can you test it?

File: tmux-cssh [hunk 1]
```diff
@@ -53,23 +57,36 @@ if [ -z "${hosts}" ]; then
 fi
 
 # Find a name for a new session
-n=0; while tmux has-session -t "${tmux_session_name}-${n}" 2>/dev/null; do n=$(($n + 1)); done
-tmux_session="${tmux_session_name}-${n}"
+n=0; while tmux has-session -t "${tmux_name}-${n}" 2>/dev/null; do n=$(($n + 1)); done
+tmux_session="${tmux_name}-${n}"
```
Comment (oryband) [inline tmux-cssh:61]: the window naming that was added in the recent commit works. one question though: why is it adding number suffix add the end? why not just use the given name?

**Review by oryband — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -27,8 +27,9 @@ Spawns multiple synchronized SSH sessions inside a tmux session.
 
 Options:
   -h                  Show help
-  -n <name>           Name of the tmux session (default: cssh)
+  -n <name>           Name of the tmux session or window (default: cssh)
```
Comment (oryband) [inline README.md:30]: also, is it possible to occupy the current window instead of creating a new one?

**Review by oryband — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (oryband) [reply README.md:30]: just a thought, this isn't critical of course

File: tmux-cssh (hunk 1 above)
Comment (jsecchiero) [reply tmux-cssh:61]: is to be consistent with session naming
https://github.com/peikk0/tmux-cssh/blob/dffe8ee34649b398ac3b9e2b12908f1d74e1cd96/tmux-cssh#L56-L57

Comment (oryband) [reply tmux-cssh:61]: ok then :) thanks for the reply and working on this issue. it's a godsend

File: README.md (hunk 2 above)
Comment (jsecchiero) [reply README.md:30]: imitate the same behavior of creating of new session i create a new window, but imho you are right is much more intuitive if use the current window. When i have free time i'll check how good is it.

File: tmux-cssh [hunk 3]
```diff
@@ -53,23 +57,36 @@ if [ -z "${hosts}" ]; then
...
+
+if [ ! -z "${tmux_attach_current_session}" ]; then
+    tmux_session=$(tmux display-message -p '#S')
+    n=0; while tmux list-windows -F "#W" | grep "${tmux_name}-${n}" 2>/dev/null; do n=$(($n + 1)); done
```
Comment (peikk0) [inline tmux-cssh]: Can you use `grep -q` instead here please?

File: tmux-cssh [hunk 4]
```diff
@@ -53,23 +57,36 @@ if [ -z "${hosts}" ]; then
...
+    tmux_window="${tmux_name}-${n}"
+    tmux_window_options="-n ${tmux_window}"
+fi
 
 # Open a new session and split into new panes for each SSH session
 for host in ${hosts}; do
     if ! tmux has-session -t "${tmux_session}" 2>/dev/null; then
         tmux new-session -s "${tmux_session}" -d "ssh ${ssh_options} ${host}"
+    elif [ ! -z "${tmux_attach_current_session}" ] && ! tmux list-windows -F "#W" | grep "${tmux_window}" >/dev/null; then
```
Comment (peikk0) [inline tmux-cssh]: Can you use `grep -q` instead here please?

File: tmux-cssh [hunk 5]
```diff
@@ -11,8 +11,9 @@ usage() {
     echo "" >&2
     echo "Options:" >&2
     echo "  -h                  Show help" >&2
-    echo "  -n <name>           Name of the tmux session (default: cssh)" >&2
+    echo "  -n <name>           Name of the tmux session or window (default: cssh)" >&2
     echo "  -o <ssh args>       Additional SSH arguments" >&2
+    echo "  -c                  Use the current tmux session and just spawn a new window instead" >&2
```
Comment (peikk0) [inline tmux-cssh]: Nit: can you move this line above the `-n` one please?

File: tmux-cssh (hunk 4 above)
Comment (peikk0) [inline tmux-cssh]: `[ ! -z` → `[ -n`

File: tmux-cssh [hunk 6]
```diff
@@ -53,23 +57,36 @@ if [ -z "${hosts}" ]; then
 fi
 
 # Find a name for a new session
-n=0; while tmux has-session -t "${tmux_session_name}-${n}" 2>/dev/null; do n=$(($n + 1)); done
-tmux_session="${tmux_session_name}-${n}"
+n=0; while tmux has-session -t "${tmux_name}-${n}" 2>/dev/null; do n=$(($n + 1)); done
+tmux_session="${tmux_name}-${n}"
+
+if [ ! -z "${tmux_attach_current_session}" ]; then
```
Comment (peikk0) [inline tmux-cssh]: `[ ! -z` → `[ -n`

File: tmux-cssh [hunk 7]
```diff
@@ -53,23 +57,36 @@ if [ -z "${hosts}" ]; then
...
+        tmux split-window -t "${tmux_session}" -t ${tmux_window} -d "ssh ${ssh_options} ${host}"
         # We have to reset the layout after each new pane otherwise the panes
         # quickly become too small to spawn any more
         tmux select-layout -t "${tmux_session}" tiled
     fi
 done
 
 # Synchronize panes by default
-tmux set-window-option -t "${tmux_session}" synchronize-panes on
+if [ "${tmux_attach_current_session}" ]; then
```
Comment (peikk0) [inline tmux-cssh]: `[ -n "${tmux_attach_current_session}" ]`

**Review by peikk0 — CHANGES_REQUESTED** [review]

**Pull request merged**